### DISCUSSION TITLE
CSL file not working with JATS format?

### DISCUSSION THREAD

**Discussion by amorris28** [quarto-dev/quarto-cli#6460] [Q&A]
### Description

I am trying to write a manuscript in Quarto and I was following [this guide](https://quarto.org/docs/authoring/footnotes-and-citations) for how to handle references. When I use `format: html` in the front matter along with `csl: nature.csl` and `bibliography: references.bib`, I get correctly formatted references at the end of my document. But if I use `format: jats`, the references are not correctly formatted--it seems like the `.csl` is being ignored. Is this supposed to work?

For what it's worth, the [JATS](https://quarto.org/docs/reference/formats/jats) section of the Quarto documentation includes a section on `csl` so I am assuming this is not working as intended.

Below is a minimal example with a single reference. I have linked to the `nature.csl` style file and provided a minimal `references.bib` below.

```qmd
---
title: "Untitled"
format: jats
editor: visual
csl: nature.csl
bibliography: references.bib
---

## Test

Test. [@darwin1964]

```

Link to the [nature.csl](https://github.com/citation-style-language/styles/blob/master/nature.csl) file.

`references.bib`:
```bib
@book{darwin1964,
  title = {On the {{Origin}} of {{Species}}: {{A Facsimile}} of the {{First Edition}}},
  shorttitle = {On the {{Origin}} of {{Species}}},
  author = {Darwin, Charles},
  year = {1964},
  month = jan,
  publisher = {{Harvard University Press}},
  address = {{Cambridge, MA}},
  abstract = {It is now fully recognized that the publication of Charles Darwin's Origin of Species in 1859 brought about a revolution in man's attitude toward life and his own place in the universe. This work is rightly regarded as one of the most important books ever published, and a knowledge of it should be part of the intellectual equipment of every educated person. The book remains surprisingly modern in its assertions and is also remarkably accessible to the layman, much more so than recent treatises necessarily encumbered with technical language and professional jargon. This first edition had a freshness and uncompromising directness that were considerably weakened in later editions, and yet nearly all available reprints of the work are based on the greatly modified sixth edition of 1872. In the only other modern reprinting of the first edition, the pagination was changed, so that it is impossible to give page references to significant passages in the original. Clearly this facsimile reprint of the momentous first edition fills a need for scholars and general readers alike. Ernst Mayr's Introduction has a threefold purpose: to list passages in the first edition that Darwin altered in later editions; to point out instances in which Darwin was clearly pioneering; and to call attention to neglected passages that show Darwin as a much deeper thinker than has been recognized. No one can fail to be impressed by the originality of Darwin's treatment and by the intellectual challenge his work presents even to the modern reader.},
  isbn = {978-0-674-63752-8}
}
```

### HTML with correct reference formatting:
![testhtml](https://github.com/quarto-dev/quarto-cli/assets/16123635/07c6b2fb-067c-46ab-8cf8-22da3b4c0d90)

### JATS with incorrect reference formatting:
![testsjats](https://github.com/quarto-dev/quarto-cli/assets/16123635/32a004b3-6214-41ad-909d-6fba79bd5786)

## Versions
```
R version 4.3.1 (2023-06-16) -- "Beagle Scouts"
quarto version 1.3.450
rstudio version 2023.06.1+524
```

**Comment by dragonstyle** [comment]
1) I did a little simple testing with our default citation format vs. a simple alternative CSL file and observed changes. Can you be more specific about what you're expecting to see that isn't present?

2) When you're previewing JATS, an XSLT stylesheet is being applied to the XML to allow you to inspect and validate the content. That said, the best to confirm that you're happy with the JATS is to look directly at the XML output, which allows you to bypass the styling, which may be laying out or altering the appearance of the document for preview purposes.

**Comment by amorris28** [comment]
When I use the default format and compare that to adding a CSL file I don't see any differences in the References lists (only in the in-text citations). What I would like to see is the References list formatted in the RStudio Viewer pane based on the CSL file. That is, with the correct information displayed (or omitted, e.g., the ISBN) and with the correct styling (italics, parentheses, spaces, etc.). So, in the case of the `nature.csl` it should be like this:

1. Darwin, C. _On the Origin of Species: A Facsimile of the First Edition._ (Harvard University Press, 1964).

Instead, when I render the document with JATS it seems to be simply concatenating all of the information in the bibtex file without any organization or formatting:

1. Charles Darwin On the Origin of Species: A Facsimile of the First EditionHarvard University PressCambridge, MA196401978-0-674-63752-8

Looking at the XML file, it seems that JATS XML formats the bibliography information without any style data. It just uses tags for `publication-type`, `source`, `publisher-name`, etc. So, maybe this is a use-case that JATS is not intended for? (But then how would a JATS XML file be used downstream, either by the user or by a publisher, to produce a formatted PDF?)

Thanks!

**Reply by dragonstyle** [reply]
In Quarto 1.4, we should be producing `element-citation` style references for JATS output - we've been doing a bunch of work on a new project type targeted at this type of output - Manuscripts (more at <https://quarto.org/docs/manuscripts/>).

In Quarto 1.3, we may not be producing these by default. You can render `--to: jats+element_citations` or set this in YAML via:

```yaml
format:
  jats:
    variant: "+element_citations"
```

And that should get you what I believe is the desirable JATS output (well structured reference data for downstream systems).

`element_citations` will not respect CSL at all.

**Reply by amorris28** [reply]
I tried the code chunk you provided and played around with the manuscript projects in Quarto 1.4. Excited to see where that project goes, but I think jats format isn't rendering the way I want so I will stick with html. Thank you!
